src/components/Editor/index.js

```diff
@@ -160,16 +164,35 @@ class Editor extends Component {
     } else {
       this.undoTimestamp = 0
     }
+
+    if (evt.keyCode === 27 && !this.state.tabGuarded) {// Esc Key
+      this.setState({tabGuarded:true})
+    }else if(!(evt.shiftKey || evt.keyCode === 27 || evt.keyCode === 9) && this.state.tabGuarded){
```

Comment:
Definitely, I'll be a little more thorough with the key clicking this go round after the change too.